File: java/cours/src/session3/TD/MotIterableAvecIterateur/MotIterableAvecIterateur.java
package session3.TD.MotIterableAvecIterateur;

public class MotIterableAvecIterateur {

	public static void main(String[] args) {
		Mot a = MotVide.VIDE.motNonVide('a');
		Mot b = MotVide.VIDE;
		for(int i = 0; i < 10000000; i++){
			b = b.concat(a);
		}
		int s = 0;
		for(int x : b){
			s = s + x;
		}
		System.out.println(s);
	}

}

File: typescript/session3/MotIterableAvecIterateurEtVisiteur/MotIterableAvecIterateur.ts
import { MotVide } from "./MotVide";
import { Mot } from "./Mot";
import { Char } from "../Char";

var a: Mot = MotVide.getInstance().mot(new Char('a'));
var b: Mot = MotVide.getInstance();
for (let i = 0; i < 10000000; i++) {
  b = b.concat(a);
}
let s = "";
for (let x in b) {
  s = s + x;
}
console.log(s);
